saltproc/serpent_depcode.py

```diff
@@ -51,118 +57,94 @@ def __init__(self,
+        include_card = [line for line in file_lines if line.startswith("include ")]
+        if not include_card:
+            raise IOError('Template file '
+                          f'{self.template_input_file_path} has no <include '
+                          '"material_file"> statements')
+        burnable_materials_path = include_card[0].split()[1][1:-1]
+        if not Path(burnable_materials_path).is_absolute():
+            absolute_path = (runtime_dir / burnable_materials_path)
         else:
-            abs_src_matfile = src_file
-            if 'mat ' not in open(abs_src_matfile).read():
-                print('ERROR: Template file %s has not include file with'
-                      ' materials description or <include "material_file">'
-                      ' statement is not appears'
-                      ' as first <include> statement\n'
-                      % (self.template_input_file_path))
-                return
+            absolute_path = Path(burnable_materials_path)
+            with open(absolute_path) as f:
+                if 'mat ' not in f.read():
+                    raise IOError('Template file '
+                                  f'{self.template_input_file_path} includes '
+                                  'no file with materials description')
         # Create data directory
-        try:
-            os.mkdir(os.path.dirname(self.iter_matfile))
-        except FileExistsError:
-            pass
+        Path.mkdir(Path(self.runtime_matfile).parents[0], exist_ok=True)
+
         # Create file with path for SaltProc rewritable iterative material file
-        shutil.copy2(abs_src_matfile, self.iter_matfile)
-        return [s.replace(src_file, self.iter_matfile) for s in template_data]
+        shutil.copy2(absolute_path, self.runtime_matfile)
+        return [line.replace(burnable_materials_path, self.runtime_matfile) for line in file_lines]
 
-    def get_nuc_name(self, nuc_code):
-        """Returns nuclide name in human-readable notation: chemical symbol
-        (one or two characters), dash, and the atomic weight. Lastly, if the
-        nuclide is in metastable state, the letter `m` is concatenated with
-        number of excited state. For example, `Am-242m1`.
+    def convert_nuclide_code_to_name(self, nuc_code):
+        """Converts Serpent2 nuclide code to symbolic nuclide name.
+        If nuclide is in a metastable state, the nuclide name is concatenated
+        with the letter `m` and the state index.
 
         Parameters
         ----------
         nuc_code : str
-            Name of nuclide in Serpent2 form. For instance, `Am-242m`.
+            Nuclide code in Serpent2 format (`47310.09c`)
 
         Returns
         -------
         nuc_name : str
-            Name of nuclide in human-readable notation (`Am-242m1`).
-        nuc_zzaaam : str
-            Name of nuclide in `zzaaam` form (`952421`).
+            Symbolic nuclide name (`Am242m1`).
```

Comment:
Are you using the nuclide code for Am-242m1? Just want to make sure you're being consistent.

Reply:
This is correct.

Reply:
Ok. That's just a very unexpected nuclide code for Am-241m1